### PR TITLE
Adding package-deps package

### PR DESCRIPTION
**Pull request by dzearing**
This separates out a package-deps package, which can generally be used to build a change description.

### CODE REVIEW

File: package-deps/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# @microsoft/package-deps
+
+`package-deps` is a general utility for building a JSON object containing the git hashes of all files used to produce a given package.
+
+This utility is useful for scenarios where you want to define a "change receipt" file to be published with a package. The file content
+and the current state of the package can be compared then to determin if the package needs to be rebuilt.
+
+Internally it uses the GIT hashes to derive the hashes for package content. This allows the process to piggyback off GIT's hashing
+optimizations, as opposed to creating a more ellaborate diffing scheme.
```
Comment (cliffkoh) [inline package-deps/README.md]: elaborate

File: package-deps/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+NOTE: GIT is required to be accessible in the command line path.
+
+## Usage
+
+
+```
+let _ = require('lodash');
+let { getPackageDeps } = require('package-deps');
```
Comment (cliffkoh) [inline package-deps/README.md]: `@microsoft` namespace?

File: package-deps/src/tests/getPackageDeps.test.ts [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+          expect(results.files[filePath])
+            .equals(expectedFiles[filePath], `path: ${filePath}`)));
+
+      } catch (e) { return done(e); }
+
+      done();
+    });
+  });
+
+  it('can exclude an added file', (done) => {
```
Comment (cliffkoh) [inline package-deps/src/tests/getPackageDeps.test.ts]: I guess you can possibly write edge case handling tests too (such as what happens when you exclude a file that does not exist, aka exclude has a random string).

File: package-deps/README.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
+# @microsoft/package-deps
+
+`package-deps` is a general utility for building a JSON object containing the git hashes of all files used to produce a given package.
```
Comment (cliffkoh) [inline package-deps/README.md]: The other thing is that when you put "deps" besides "package", people would typically interpret deps as "npm deps", so why not just package-hash and drop "deps" completely?
